grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java

```diff
@@ -75,27 +88,35 @@ static RemoteGraknTx create(RemoteGraknSession session, GraknTxType txType) {
 
     @Override
     public EntityType putEntityType(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ENTITY);
     }
 
     @Override
     public <V> AttributeType<V> putAttributeType(Label label, AttributeType.DataType<V> dataType) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ATTRIBUTE, var("x").datatype(dataType));
     }
 
     @Override
     public Rule putRule(Label label, Pattern when, Pattern then) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, RULE, var("x").when(when).then(then));
     }
 
     @Override
     public RelationshipType putRelationshipType(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, RELATIONSHIP);
     }
 
     @Override
     public Role putRole(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ROLE);
+    }
+
+    private <X extends Concept> X putSchemaConcept(Label label, Schema.MetaSchema meta, VarPattern ... optionalVars){
+        List<VarPattern> vars = new ArrayList<>();
+        vars.add(var("x").label(label).sub(meta.getLabel().getValue()));
+        vars.addAll(Arrays.asList(optionalVars));
+        queryRunner().run(Graql.withoutGraph().define(vars));
```

Comment:
You can skip `withoutGraph` if you like and do `Graql.define(vars)`

Reply:
What's the point of the `withoutGraph` method then? I.e. what's the diff?

Reply:
Basically as a means to get at a `QueryBuilder`, since some methods might accept a `QueryBuilder` and then build queries with it. The only example I can find in the code is `Migrator`, which gets a `QueryBuilder` so it can toggle inference off.